src/charm.py

```diff
@@ -143,6 +144,38 @@ def snap_path(self) -> Optional[str]:
 
         return self._snap_path
 
+    @property
+    def snap_channel(self) -> str:
+        """Get the channel for exporter snap.
+
+        The channel is determined by the controller version that the charm is deployed
+        under. In case the controller version is less than 2.9, the snap channel is set
+        to 2.8/stable. Otherwise, the snap channel is set to 2.9/stable.
```

Comment:
I think we could move it one more step ahead and check min version, because I do not think that 2.8/stable will work with controller 2.6. Maybe there is no such controller, but since we have function and logic to check it, it will be good if we do so.

Reply:
2.6 is actually supported by 2.8/stable and it is the minimum version that has been tested. Good suggestion on checking the minimal supported version. I have made the changes accordingly.